docs/release-notes.md

```diff
@@ -32,6 +32,21 @@ updates as well.
   2.9.10 has an infinite loop in a certain end-of-file situation." Fix
   by updating a dependency in documentation generation.
 
+### Zephyr-Rtos compatibility
+- for compatibility with zephyr 2.2.0 the following commits need to be reverted:  
+    f4a9a9d  
```

Comment:
While I think it's very laudable to try and help users with coping with the lack of compatibility between Zephyr and MCUBoot releases, this might just not scale in the future. It's a bit fragile to have to depend on a list of commits that need to be reverted, and that might introduce other issues.
Can I suggest another approach? Instead of trying to make a particular release of MCUBoot compatible with a Zephyr one, and since this is the release notes for MCUBoot, how about we just document which Zephyr SHA this MCUBoot release is guaranteed to work with, and leave it to users to figure out what they need to do if they want to actually uses tagged releases from both projects?

As we've discussed multiple times with @nvlsianpu, @utzig and @d3zd3z, there are enough dependencies in a bootloader that builds against different RTOS to make it very hard to maintain compatibility in certain periods, like the current one where Zephyr is simply changing too much in very little time. What I see as future-proof approaches to this problem could be:

1. Remove the dependency to Zephyr, Mynewt and RIOT, and instead bundle all the code necessary to build MCUBoot in this repository
2. Not release MCUBoot at all, instead just keeping it rolling, and relying on the different RTOS to signal which particular MCUBoot SHA is compatible with each SHA in their respective histories
3. Start maintaining backwards compatibility for the RTOS that MCUBoot supports, including (a lot of) conditional compilation based on which particular release or SHA of the RTOS we are building against
4. Release, but do the minimum to document which SHA of each RTOS is supported with this release

Let me know your thoughts.

Reply:
> 1. Remove the dependency to Zephyr, Mynewt and RIOT, and instead bundle all the code necessary to build MCUBoot in this repository

I think this would be even harder to scale. We already have one target that builds this way. It pulls in 5 external repos in order to get enough dependencies, and has its own directory with 13k lines of code in it. The Zephyr-specific directory is about 1/3 of this, and builds with a normal Zephyr. I think no matter the issues we find with Zephyr, they are smaller than the issues we would find trying to build bare-bones.

However, I don't think this problem is particularly related to it being a bootloader, either. The bootloader is actually a pretty simple program. Given how fast Zephyr's API surface changes, I would guess more significant applications would encounter this even more so.

> 2. Not release MCUboot at all.

People seem to like seeing releases, even though, as you point out, they really aren't that helpful.

> 3. ... backward compatibility ... conditional compilation.

At least now this is possible, but it gets fairly messy.

> 4. Release, but do the minimum to document which SHA of each RTOS is supported with this release.

I think people do like to use labeled releases of Zephyr as well. On possible hybrid with this, would be to release mcuboot against master of Zephyr, and then make branches in the Zephyr-fork of MCUboot to backport to each release that is desired. I'm just not sure who has time to do this work.

Regardless, we'll need to be clear about what we do. We've stated before (not sure if it made the docs) that mcuboot/master should work with zephyr/master, and mcuboot's releases should work with the latest release of Zephyr.

One other idea, is we could make something like a v1.6.0-zephyr-2.2.0 release of MCUboot as a branch already made to work with Zephyr.

Reply:
> On possible hybrid with this, would be to release mcuboot against master of Zephyr, and then make branches in the Zephyr-fork of MCUboot to backport to each release that is desired. I'm just not sure who has time to do this work.

I think branches for specific Zephyr versions should live here and be *downstreamed*.

> One other idea, is we could make something like a v1.6.0-zephyr-2.2.0 release of MCUboot as a branch already made to work with Zephyr.

This to me sounds like the best option, we could do releases against `master` and branches against the *latest* stable (or LTS) release.

Reply:
I will reply on the main conversation, because I think that keeping branches like the one proposed might be too much overhead.